homeassistant/components/comelit/cover.py

```diff
@@ -0,0 +1,101 @@
+"""Support for covers."""
+from __future__ import annotations
+
+from typing import Any
+
+from aiocomelit import ComelitSerialBridgeObject
+from aiocomelit.const import COVER, COVER_CLOSE, COVER_OPEN, COVER_STATUS
+
+from homeassistant.components.cover import CoverDeviceClass, CoverEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import ComelitSerialBridge
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Comelit covers."""
+
+    coordinator: ComelitSerialBridge = hass.data[DOMAIN][config_entry.entry_id]
+
+    # Use config_entry.entry_id as base for unique_id because no serial number or mac is available
+    async_add_entities(
+        ComelitCoverEntity(coordinator, device, config_entry.entry_id)
+        for device in coordinator.data[COVER].values()
+    )
+
+
+class ComelitCoverEntity(CoordinatorEntity[ComelitSerialBridge], CoverEntity):
+    """Cover device."""
+
+    _attr_device_class = CoverDeviceClass.SHUTTER
+    _attr_has_entity_name = True
+
```

Comment:
```suggestion
    _attr_name = None

```

Reply:
Since the device you set as device info has the same name as this entity. This will name the entity to the device.